synapse/event_auth.py

```diff
@@ -137,6 +137,22 @@ def check_auth_rules_for_event(
     Raises:
         AuthError if the checks fail
     """
+    # Before we get too far into event auth, validate that the event is even
+    # valid enough to be used
+    if event.type == EventTypes.PowerLevels:
+        # If applicable, validate that the known power levels are integers
+        if room_version_obj.msc3667_int_only_power_levels:
+            for k, v in event.content.items():
+                if k in ["events", "notifications", "users"]:
+                    if type(v) is not dict:
```

Comment:
One usually does `isinstance(v, dict)`, but it doesn't really matter in this case (`isinstance` correctly handles subclasses and the like)